src/app/css-variables.ts

```diff
@@ -1,56 +1,74 @@
 import { settingsSelector } from 'app/dim-api/selectors';
+import { deepEqual } from 'fast-equals';
 import { isPhonePortraitSelector } from './shell/selectors';
-import { observeStore } from './utils/redux';
+import { StoreObserver } from './store/observerMiddleware';
 
 function setCSSVariable(property: string, value: { toString: () => string }) {
   if (value) {
     document.querySelector('html')!.style.setProperty(property, value.toString());
   }
 }
 
+export function createItemSizeObserver(): StoreObserver<number> {
+  return {
+    id: 'item-size-observer',
+    getObserved: (rs) => settingsSelector(rs).itemSize,
+    sideEffect: ({ previous, current }) => {
+      if (previous !== current) {
+        setCSSVariable('--item-size', `${Math.max(48, current)}px`);
+      }
+    },
+  };
+}
+
+export function createThemeObserver(): StoreObserver<{ theme: string; isPhonePortrait: boolean }> {
+  return {
+    id: 'theme-observer',
+    equals: deepEqual,
+    getObserved: (rs) => ({
+      theme: settingsSelector(rs).theme,
+      isPhonePortrait: isPhonePortraitSelector(rs),
+    }),
+    sideEffect: ({ previous, current }) => {
+      // TODO (ryan): Hoist the feature flag check up to where the observer is created
+      // Set a class on the body to control the theme. This must be applied on the body for syncThemeColor to work.
+      if ($featureFlags.themePicker && previous.theme !== current.theme) {
+        const themeClass = `theme-${current.theme}`;
+        document.body.className = themeClass;
+        syncThemeColor(current.isPhonePortrait);
+      }
+    },
+  };
+}
+
+export function createTilesPerCharColumnObserver(): StoreObserver<{
+  charCol: number;
+  charColMobile: number;
+  isPhonePortrait: boolean;
+}> {
+  return {
+    id: 'tiles-per-char-column-observer',
+    equals: deepEqual,
+    getObserved: (rs) => ({
+      charCol: settingsSelector(rs).charCol,
+      charColMobile: settingsSelector(rs).charColMobile,
+      isPhonePortrait: isPhonePortraitSelector(rs),
+    }),
+    sideEffect: ({ current }) => {
+      setCSSVariable(
+        '--tiles-per-char-column',
+        current.isPhonePortrait ? current.charColMobile : current.charCol,
+      );
+      syncThemeColor(current.isPhonePortrait);
+    },
+  };
```

Comment:
You could use a selector here:

```suggestion
  return {
    id: 'tiles-per-char-column-observer',
    equals: deepEqual,
    getObserved: (rs) => ({
      tilesPerCharColumn: isPhonePortraitSelector(rs) ? settingsSelector(rs).charColMobile : settingsSelector(rs).charCol,
      isPhonePortrait: isPhonePortraitSelector(rs),
    }),
    sideEffect: ({ current }) => {
      setCSSVariable(
        '--tiles-per-char-column',
        current.tilesPerCharColumn,
      );
      syncThemeColor(current.isPhonePortrait);
    },
  };
```

I'm not convinced syncThemeColor needs to be a part of this either - it should be independently reacting to `isPhonePortrait` changes.

Reply:
I'll commit this after I figure out the portrait thing